build-aux/main.mk

```diff
@@ -124,3 +124,25 @@ PHONY: chart
 
 boguschart_dir = build-output/chart-2.0.0-bogus_7.0.0-bogus.d
 boguschart_tgz = $(patsubst %.d,%.tgz,$(boguschart_dir))
+
+# YAML manifests
+build-output/yaml-%: $(shell find $(CURDIR)/manifests/emissary/ -type d -o -name '*.yaml.in') $(var.)DEV_REGISTRY $(var.)RELEASE_REGISTRY
+ifeq ($(CI),)
+	rm -rf $@
+else
+	@if test -d $@; then \
+	  echo 'This should not happen in CI: $@ should not need to change' >&2; \
+	  echo 'Files triggering the change are: $?' >&2; \
+	  exit 1; \
+	fi
+endif
+	mkdir -p $@
+	$(foreach src,$(filter %.yaml.in,$^),$(foreach dst,$(patsubst $(CURDIR)/manifests/emissary/%.yaml.in,$@/%.yaml,$(src)),\
```

Comment:
😂  Nobody other than you is ever going to dare read this thing, much less edit it. Especially given that your second `foreach` isn't even a loop, unless I'm badly misreading something, since the first `foreach` handles all of that. Why not just do it in the shell, so that it's more clear where the iteration is happening and what the loop values are?

```shell
for src in $(filter %.yaml.in,$^); do \
  dst=$(echo "$$src" | sed -e 's!^$(CURDIR)/manifests/emissary/!$@!' -e 's!\.in$$!!') ;\
  if [[ "$$dst" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+|-ea)?$$ ]]; then \
    registry=$(RELEASE_REGISTRY) ;\
  else \
    registry=$(DEV_REGISTRY) ;\
  fi ;\
  sed -e 's/\$$version\$$/$*/g' -e 's,\$$imageRepo\$$,'"$${registry}"'/emissary,g' <$(src) >$(dst) ;\
done
```

Less elegant Make-fu, but probably easier for most to reason about...

Reply:
I'm certain that _somewhere_ where I do this (maybe not in emissary.git... maybe telepresence.git? or getambassador.io.git?) I have a comment explaining it.

```make
$(foreach key,val,\
    body)
```
is just Make's funny way of spelling the Lisp
```lisp
(let ((key val))
    body)
```

Reply:
The reason I prefer to do things with Make's loops is so that I can use make --debug to see what it's going to do before I actually run it.

Reply:
> "Make expressions are just Lisp with dollar signs, commas, and backslashes."

...that doesn't make Lisp any easier for non-Lisp speakers. 😂